### PR TITLE
add '-f' flag to append arbitrary compiler flags

### PR DESCRIPTION
**Pull request by ProducerMatt**
There's some flags I want to specify beyond -s or -l flags, and this lets me do that. :)

### CODE REVIEW

File: c99sh [hunk 1]
```diff
@@ -193,13 +195,13 @@ done
 [ $v -ge 1 ] && set -x
 case "${name:0:3}" in
   c11|C11)
-    "${CC-cc}"   -std=gnu11   -o "$x" "${flags[@]}" -x c   "${logic[@]}" 1>&2
+    "${CC-cc}"   -std=gnu11   -o "$x" "${flags[@]}" -x c   "${logic[@]}" ${CF[@]} 1>&2
```
Comment (RhysU) [inline c99sh:198]: Should this have double quotes around it?

**Comment by RhysU** [comment]
I appreciate the CR!
See comments about re: adding double quotes.
Please feel free to add this contribution to the "Credits" in the README.

File: c99sh (hunk 1 above)
Comment (ProducerMatt) [reply c99sh:198]: I originally did that, but found that multiple flags would get quoted in the final shell command. So if you did `c99sh -f "-O0 -fsummon-cthulu"` then cc would get `-O0\ -fsummon-cthulu`, so one long flag with escaped spaces. I guess it's got something to do with the processing that `$logic` and `$flags` receive that `$CF` doesn't. Anyways it's working correctly without quotes, but I don't know sh well so if you see an improvement let me know :)

**Comment by ProducerMatt** [comment]
I added the credit and responded re the quotes.

**Comment by RhysU** [comment]
Hi @ProducerMatt, did you see the question about if arbitrary flags can be accompllished already with the rcfile?

**Comment by ProducerMatt** [comment]
I don't see any question regarding that, maybe your comment didn't post?

If the question was "why not use the config file to pass custom flags", I like examining compiler behavior, and -f allows me to set per-file flags without needing to retype the entire cc command, since c99sh will add the other default flags for me.

**Comment by RhysU** [comment]
Does `CFLAGS=-fstuff ...` at the beginning of your command accomplish what you want?

**Comment by ProducerMatt** [comment]
If compiler flags are set left-to-right then $CFLAGS could get overridden by $LOGIC. $CF being all the way to the right means it can't be overridden by some default that's set somewhere else. Or so goes my line of thinking.

I know this is a very specific (and somewhat pedantic) feature, so I wouldn't be offended if you decided it was unnecessary ;)

**Comment by RhysU** [comment]
Nah, I want to land this. Just trying to see if there's something slick that'll accomplish what you want.

**Comment by RhysU** [comment]
Hey @ProducerMatt , would #40 do what you want?  Specifically, either the proposed `-F` or the proposed `-L`?  Those hook into CFLAGS/LDFLAGS but they change none of the later logic, which is a little cleaner.

**Comment by ProducerMatt** [comment]
That would work if it can take multiple flags. Like `-F '-boo -far'`

**Comment by RhysU** [comment]
Multiple like`-Ffoo` then `-Fbar`  should work.

**Comment by RhysU** [comment]
@ProducerMatt, does #40 do what you want with `-FO0 -Ffsummon-cthulu`?

**Comment by ProducerMatt** [comment]
Yes I believe it does :)

**Pull request merged**